samples/nrf9160/nrf_cloud_mqtt_multi_service/src/connection.c

```diff
@@ -656,7 +656,8 @@ static int connect_cloud(void)
 	}
 
 	LOG_INF("Connected to nRF Cloud");
-	return 0;
+
+	return err;
```

Comment:
`err` cannot be nonzero at this location, but I do see it is a little confusing to have `int err` declared at the start of the function

Perhaps it would be better to move the err declaration to `int err = nrf_cloud_connect();` inside the while loop

Honestly, it would be worth looking at removing the while loop in `connect_cloud` altogether. I don't think it is really needed, since `connection_management_thread_fn` already has persistence built in. But that should probably be a separate PR. I'll write a ticket for it.

Reply:
I added the while-loop investigation to IRIS-4741

Reply:
OK, leaving it for now